hashicorp/dev-portal
# feat: add tutorial and collection cards

## ✅ "Ready for Review" checklist




---

## 🔗 Relevant links

- [Preview - `CollectionCard`][swingset-collection-card] 🔎
- [Preview - `TutorialCard`][swingset-tutorial-card] 🔎
- [Asana task](https://app.asana.com/0/1201987349274776/1201654639085737/f) 🎟️

## 🗒️ What

This PR implements `components/tutorial-card` & `components/collection-card`. It also replaces any existing stubbed implementations of these components with the new implementations.

## 🤷 Why

To finish building out the tutorial views.

## 🛠️ How

- Sets up a `components/tutorial-collection-cards` folder for shared elements used in both cards
- Composes those shared elements to build `components/tutorial-card`
- Composes those shared elements to build `components/collection-card`

## 📸 Screenshots

<details>

<summary>Tutorial Card</summary>

Design:

<img width="985" alt="CleanShot 2022-04-21 at 15 59 02@2x" src="https://user-images.githubusercontent.com/4624598/164542793-290af1db-8267-401b-9405-2893448400ab.png">

Implementation:

<img width="792" alt="tutorial-card" src="https://user-images.githubusercontent.com/4624598/164541756-9d105fa5-16a5-4571-9c74-d81feecc4991.png">

</details>

<details>

<summary>Collection Card</summary>

Design:

<img width="962" alt="CleanShot 2022-04-21 at 15 59 44@2x" src="https://user-images.githubusercontent.com/4624598/164542906-d4cca7bf-e38d-44de-8888-30f51f9bf901.png">

Implementation: 

<img width="788" alt="collection-card" src="https://user-images.githubusercontent.com/4624598/164541829-09f03d94-790d-4660-87a7-a664da6b6ab7.png">

</details>

## 🧪 Testing

- Visit Swingset, and review the [`Collection Card`][swingset-collection-card] & [`Tutorial Card`][swingset-tutorial-card] pages, and make sure things look correct
    - Note that the `description` on both types of cards is truncated to 3 lines using CSS.
    - [x] Truncation should function as expected in all major browsers (tested in Safari, Firefox, & Chrome)
- Visit a tutorial view that shows "Featured Collections", such as [the Waypoint install tutorial][waypoint-install-tutorial]
    - [ ] Ensure the "Also appears in" cards at the bottom of the page look correct.
- Visit [the home page][home-page]
    - [ ] Ensure the tutorial cards partway down the page look as expected.
    - Note: the data for these cards is stubbed in placeholder. So it is unlikely to look correct.

> Note: If you want, you can review the `CollectionCard` & `TutorialCard` components [in Storybook][storybook]. I found mobile behaviour & truncation a little easier to test.

## 💭 Anything else?

- This PR does _not_ try to use the new components in new places. It only replaces existing stubbed implementations. These components will be used on our tutorials views in subsequent PRs (which will large be constructing grid layouts with these cards).
- Super minor nitpick: The `badge` size is just _slightly_ off. Fixing this would require changes to the `badge` component, and I think would implicate the `badge` component's current `display: inline-flex`. This is very much a pixel-pushing issue, like 2px of extra whitespace, so _maybe_ will try to tweak this before June 20, but definitely need to understand the best way forward to do so.

[home-page]: https://dev-portal-git-zsadd-tutorial-collection-cards-hashicorp.vercel.app
[waypoint-install-tutorial]: https://dev-portal-git-zsadd-tutorial-collection-cards-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-install
[swingset-collection-card]: https://dev-portal-git-zsadd-tutorial-collection-cards-hashicorp.vercel.app/swingset/components/collectioncard
[swingset-tutorial-card]: https://dev-portal-git-zsadd-tutorial-collection-cards-hashicorp.vercel.app/swingset/components/tutorialcard
[storybook]: https://dev-portal-with-storybook.vercel.app/?path=/story/components-tutorialcard--playground
